File: cpp/hot100/8.最长无重复子串代码问题.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0, right = 0;
        int n = s.size();
        int ans = 0;
        unordered_set<char> map;
        for(; right < n; ++right){
            char ch = s[right];
            while(map.count(ch)){
                char c = s[left++];
                map.erase(c);
            }
            map.insert(ch);
            ans = max(ans, right - left + 1);
        }
        return ans;
    }
};

File: python/hot100/8.最长无重复子串代码问题.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        occ=set()
        left,right,ans=0,0,0
        n=len(s)
        for right in range(0,n):
            ch=s[right]
            while ch in occ:
                occ.remove(s[left])
                left+=1
            occ.add(ch)
            ans=max(ans,right-left+1)
        return ans